azure/azure_kv.go

```diff
@@ -98,8 +115,14 @@ func (az *azureSecrets) GetSecret(
 	}
 
 	t := func() (interface{}, bool, error) {
-		secretResp, err := az.kv.GetSecret(ctx, az.baseURL, secretID, "")
+		// passing empty version to always get the latest version of the secret.
+		secretResp, err := az.kv.GetSecret(ctx, secretID, "", nil)
 		if err != nil {
+			// don't retry if the secret key was not found
+			responseError, ok := err.(*azcore.ResponseError)
+			if ok && responseError.StatusCode == 404 {
+				return nil, false, secrets.ErrSecretNotFound
```

Comment:
Can you return this error instead. - `ErrInvalidSecretId` since all the other providers are returning it when a secret is not found.

Reply:
@adityadani Sorry for the delay. I'll update the PR this week.